sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala

```diff
@@ -291,37 +289,57 @@ class Dataset[T] private[sql](
         }
       }
 
+      rows = rows.map {
+        _.zipWithIndex.map { case (cell, i) =>
+          if (truncate > 0) {
+            StringUtils.leftPad(cell, colWidths(i))
+          } else {
+            StringUtils.rightPad(cell, colWidths(i))
+          }
+        }
+      }
```

Comment:
We should do this in `showString`? And we can move `minimumColWidth` into the `showString` in that case?

Reply:
Do this in getRows here is to reuse the truncate logic. I think its the same problem with we discuss in here:
![image](https://user-images.githubusercontent.com/4833765/40711061-d0762762-642c-11e8-9249-2465ee3e2536.png)
If we don't need truncate, we can move the logic and `minimumColWidth` in `showString`. I would like to hear your suggestions.

Reply:
Oh, I see, the pad rows only useful in console mode, so not need in html code. I'll do this ASAP.

Reply:
Thanks, done in 9c6b3bb.